server/src/main/java/timely/store/DataStoreImpl.java

```diff
@@ -215,6 +197,48 @@ public void run() {
         }
     }
 
+    private static final String DEFAULT_AGE_OFF_KEY = "default";
+    private static final String AGE_OFF_PREFIX = "ageoff-";
+    private static final EnumSet<IteratorScope> AGEOFF_SCOPES = EnumSet.allOf(IteratorScope.class);
+
+    private void removeAgeOffIterators(Connector con, String tableName) throws Exception {
+        Map<String, EnumSet<IteratorScope>> iters = con.tableOperations().listIterators(tableName);
+        iters.forEach((name, scopes) -> {
+            if (name.startsWith(AGE_OFF_PREFIX)) {
+                try {
+                    con.tableOperations().removeIterator(tableName, name, AGEOFF_SCOPES);
+                } catch (Exception e) {
+                    // do nothing
+                }
+            }
+        });
+    }
+
+    private void applyAgeOffIterator(Connector con, String tableName, Configuration config) throws Exception {
+        int priority = 100;
+        for (Entry<String, Integer> e : config.getMetricAgeOff().entrySet()) {
+            if (e.getKey().equals(DEFAULT_AGE_OFF_KEY)) {
+                continue;
+            }
+            String ageoff = Long.toString(e.getValue() * 86400000L);
+            Map<String, String> ageOffOptions = new HashMap<>();
+            ageOffOptions.put("ttl", ageoff);
+            ageOffOptions.put(MetricAgeOffFilter.METRIC, e.getKey());
+            IteratorSetting ageOffIteratorSettings = new IteratorSetting(priority++, AGE_OFF_PREFIX + e.getKey(),
+                    MetricAgeOffFilter.class, ageOffOptions);
+            connector.tableOperations().attachIterator(tableName, ageOffIteratorSettings, AGEOFF_SCOPES);
+        }
+        // Apply the default age off filter last
+        if (config.getMetricAgeOff().containsKey(DEFAULT_AGE_OFF_KEY)) {
+            String ageoff = Long.toString(config.getMetricAgeOff().get(DEFAULT_AGE_OFF_KEY) * 86400000L);
+            Map<String, String> ageOffOptions = new HashMap<>();
+            ageOffOptions.put("ttl", ageoff);
+            IteratorSetting ageOffIteratorSettings = new IteratorSetting(priority, AGE_OFF_PREFIX + "default",
+                    AgeOffFilter.class, ageOffOptions);
+            connector.tableOperations().attachIterator(tableName, ageOffIteratorSettings, AGEOFF_SCOPES);
+        }
```

Comment:
Thinking about this more, I might decline this and change the implementation. This implementation could potentially incur a large stack of iterators and the default ageoff must be larger than all of the other ageoff values. In an alternate implemenation, I could serialize the ageoff settings map to a single iterator (not a stack) and use the map internally in the iterator. Transforming the map to an FST may (or not) be useful.